### DISCUSSION TITLE
Sticky header makes usage in layouts impossible

### DISCUSSION THREAD

**Discussion by lukasbash** [KevinVandy/mantine-react-table#353] [Q&A]
### mantine-react-table version

beta

### react & react-dom versions

v18.2.0

### Describe the bug and the steps to reproduce it

Given the table is inside some basic app shell layout with a global app bar, and maybe some nested menu bars, the table always sets `100vh` as a height reference for calculations. I know that this is mentioned explicitly in the docs, but this makes the usage of a table with any container around it (above or below) impossible, doesn't it?
Also for the calculations, fixed values for e.g. the top toolbar height (80px) are used.
Is there a way to use the sticky header feature considering 100% height of the parent?

### Minimal, Reproducible Example - (Optional, but Recommended)

-

### Screenshots or Videos (Optional)

-

### Do you intend to try to help solve this bug with your own PR?

No, because I do not have time to dig into it

### Terms

- [X] I understand that if my bug cannot be reliably reproduced in a debuggable environment, it will probably not be fixed and this issue may even be closed.

**Comment by lukasbash** [comment]
Interesting fact I discovered by accident: If I use an empty bottom toolbar, it suddenly just becomes usable.
Sticky header now works as expected without causing an oversized overflow (not using 100vh somehow?).
The only problem is that the bottom bar falls back to the minimum height of 3rem.

Thought it's worth to mention this.

**Comment by KevinVandy** [comment]
The default heights are just a default. Specify your own table heights to your heart's desire with CSS.

**Comment by lukasbash** [comment]
> The default heights are just a default. Specify your own table heights to your heart's desire with CSS.

Tried to do so in combination with the bottom toolbar, but I cannot really get it working without being visible (at some point this feels like a hack) and the sticky header.
It would be awesome if it just works without using the bottom toolbar at all. Just by using 100% height while overriding the vh default, it doesn't work.

**Comment by Pirulax** [comment]
I've wasted many hours of my life, and still couldn't figure out how to make it fill all the available space.
The following does nothing:
```
mantineTableContainerProps: {
  height: '100%'
},
```ts
Even if I calculate the height using some CSS:
```ts
mantineTableContainerProps: {
   height: 'calc(100vh - var(--app-shell-header-height, 0px) - var(--app-shell-footer-height, 0px) - var(--app-shell-padding) * 2)'
},
```
Now, what did work was:
```ts
mantineTableContainerProps: { 
  style: {
    height: 'calc(100vh - var(--app-shell-header-height, 0px) - var(--app-shell-footer-height, 0px) - var(--app-shell-padding) * 2)'
  }
},
```
But, it STILL had overflow, so I had to use some magic value to make it disappear:
```ts
mantineTableContainerProps: { 
  style: {
    height: 'calc(100vh - var(--app-shell-header-height, 0px) - var(--app-shell-footer-height, 0px) - var(--app-shell-padding) * 2  - 114px)'
  }
},
```
There must be some better solution for this....

**Reply by ZuebeyirEser** [reply]
By the way, you can use the  CSS [clamp() ](https://developer.mozilla.org/en-US/docs/Web/CSS/clamp) function to make the height responsive.

**Reply by Pirulax** [reply]
Could you please elaborate with an example?
Thank you
